subsys/bluetooth/controller/hci_driver.c

```diff
@@ -192,22 +193,43 @@ static void data_packet_process(u8_t *hci_buf)
 	bt_recv(data_buf);
 }
 
-static void event_packet_process(u8_t *hci_buf)
+static bool event_packet_is_discardable(const u8_t *hci_buf)
 {
 	struct bt_hci_evt_hdr *hdr = (void *)hci_buf;
-	struct net_buf *evt_buf;
 
-	if (hdr->evt == BT_HCI_EVT_CMD_COMPLETE ||
-	    hdr->evt == BT_HCI_EVT_CMD_STATUS) {
-		evt_buf = bt_buf_get_cmd_complete(K_FOREVER);
-	} else {
-		evt_buf = bt_buf_get_rx(BT_BUF_EVT, K_FOREVER);
-	}
+	switch (hdr->evt) {
+	case BT_HCI_EVT_LE_META_EVENT: {
+		struct bt_hci_evt_le_meta_event *me = (void *)&hci_buf[2];
 
-	if (!evt_buf) {
-		BT_ERR("No event buffer available");
-		return;
+		switch (me->subevent) {
+		case BT_HCI_EVT_LE_ADVERTISING_REPORT:
+		case BT_HCI_EVT_LE_EXT_ADVERTISING_REPORT:
+			return true;
+		default:
+			return false;
+		}
+	}
+	case BT_HCI_EVT_VENDOR:
+	{
+		u8_t subevent = hci_buf[2];
+
+		switch (subevent) {
+		case HCI_VS_SUBEVENT_CODE_QOS_CONN_EVENT_REPORT:
+			return true;
```

Comment:
Should we add build asserts to ensure that the discardable buffer size is large enough? 
We could consider doing that for all events.
```suggestion
			BUILD_ASSERT(sizeof(BT_HCI_EVT_HDR_SIZE + 1 +HCI_VS_SUBEVENT_CODE_QOS_CONN_EVENT_REPORT)
			< CONFIG_BT_DISCARDABLE_BUF_SIZE);
			return true;
```

Reply:
If you enable assert then net_buf_add_mem will assert if writing past the buffer.
I think that is good enough. It should be caught quickly enough.
The default for the discardable buffer size is the largest advertising report.